bindings/azure/eventhubs/metadata.yaml

```diff
@@ -0,0 +1,165 @@
+    metadata:
+      - name: connectionString
+        required: true
+        sensitive: true
+        description: |
+          Connection string for the Event Hub or the Event Hub namespace.
+          Mutually exclusive with eventHubNamespace field.
+          Required when not using Azure AD Authentication.
+        example: '""Endpoint=sb://{EventHubNamespace}.servicebus.windows.net/;SharedAccessKeyName={PolicyName};SharedAccessKey={Key};EntityPath={EventHub}""'
+      - name: eventHub
+        type: string
+        required: false # Optional when a connectionString is provided
+builtinAuthenticationProfiles:
+  - name: "azuread"
+    metadata:
+      # Either connectionString or eventHubNamespace must be defined, but not both
+      - name: eventHubNamespace
+        type: string
+        required: true
+        description: |
+          The Event Hub Namespace name.
+          Mutually exclusive with connectionString field.
+          Required when using Azure AD Authentication.
+        example: '"namespace"'
+      - name: eventHub
+        type: string
+        required: true # required for AD
+        description: |
+          The name of the Event Hubs hub (“topic”). Required if using Azure AD
+          authentication or if the connection string doesn’t contain an
+          EntityPath value.
+        example: '"mytopic"'
+metadata:
+  # Input and output metadata
+
+  # Entity management is only possible when using Azure AD
+  - name: enableEntityManagement
+    type: bool
+    required: false
+    default: "false"
+    description: |
+      Boolean value to allow management of the EventHub namespace and
+      storage account.
+  - name: messageRetentionInDays
+    type: number
+    required: false
+    default: "1"
+    example: "90"
+    description: |
+      Number of days to retain messages for in the newly created Event
+      Hub namespace. Used only when entity management is enabled.
+  - name: partitionCount
+    type: number
+    required: false
+    default: "1"
+    example: "3"
+    description: |
+      Number of partitions for the new Event Hub namespace. Used only when
+      entity management is enabled.
+  - name: partitionId
+    type: string
+    required: false
+    description: |
+      DEPRECATED.
+  - name: resourceGroupName
+    type: string
+    required: false
+    description: |
+      Name of the resource group the Event Hub namespace is part of.
+      Required when entity management is enable.
+    example: '"test-rg"'
+
+
+  # Input-only metadata
+  # consumerGroup is an alias for consumerId, if both are defined consumerId takes precedence.
+  - name: consumerId
+    type: string
+    required: true # consumerGroup is an alias for this field, let's promote this to default
+    bindings:
+      input: true
+      output: false
+    description: |
+      The name of the Event Hubs Consumer Group to listen on
+    example: '"group1"'
+  - name: consumerGroup
```

Comment:
IMHO this is confusing. We should mark one of them as deprecated (see above about adding it to the schema)?

Reply:
Agreed. But how do we handle aliases? By deprecating one an stating their alias?

Reply:
Aliases should just not be included in the metadata.yaml. Just add the "latest and greatest".

Reply:
Ok. I marked as deprecated. Now, the I cannot bluntly remove it as that would be a breaking change -- and ` make check-component-metadata would complain about an undefined property ;)